In [1]:
# Import pandas
import pandas as pd

In [4]:
# Dataset URL
dataset_url = "https://raw.githubusercontent.com/GeethaGunasekaran1/Dataset_rep/refs/heads/main/crowdfunding.csv"

# Read the dataset from GitHub
mobile_data = pd.read_csv(dataset_url)

# Folder where you want to save the dataset
save_path = r"C:\Users\Poornima\OneDrive\Desktop\VS_Sample\venv\Assignments\Assignments7_CCSP\crowdfunding_dataset.csv"

# Save the dataset
mobile_data.to_csv(save_path, index=False)

print("✅ Dataset downloaded successfully!")
print("✅ Dataset saved at:")
print(save_path)

✅ Dataset downloaded successfully!
✅ Dataset saved at:
C:\Users\Poornima\OneDrive\Desktop\VS_Sample\venv\Assignments\Assignments7_CCSP\crowdfunding_dataset.csv


In [ ]:
#Step 1: Load Dataset
#We are opening the data file and looking at the first few rows.
# Load dataset
df = pd.read_csv("crowdfunding_dataset.csv")

# Show first 5 rows
print(df.head())

   Unnamed: 0 CampaignID  GoalAmount  RaisedAmount  DurationDays  NumBackers  \
0           0  CID_00000       76159  73518.028955            15        1903   
1           1  CID_00001       69394  60068.683288            49         469   
2           2  CID_00002       18652  22958.153470            58         148   
3           3  CID_00003       19487  24229.048390            39         780   
4           4  CID_00004       26734  17984.630311            68        1622   

  Category LaunchMonth    Country Currency  OwnerExperience VideoIncluded  \
0     Film   September  Australia      GBP               11           Yes   
1      Art     October  Australia      GBP                7           Yes   
2     Film    February        USA      EUR               17            No   
3     Film     October    Germany      AUD               10            No   
4    Games   September  Australia      AUD               12           Yes   

   SocialMediaPresence  NumUpdates  IsSuccessful  
0    

In [ ]:
#Step 2: Check Data Information
print(df.info()) 
#info() → shows missing values and column types


<class 'pandas.DataFrame'>
RangeIndex: 100000 entries, 0 to 99999
Data columns (total 15 columns):
 #   Column               Non-Null Count   Dtype  
---  ------               --------------   -----  
 0   Unnamed: 0           100000 non-null  int64  
 1   CampaignID           100000 non-null  str    
 2   GoalAmount           100000 non-null  int64  
 3   RaisedAmount         100000 non-null  float64
 4   DurationDays         100000 non-null  int64  
 5   NumBackers           100000 non-null  int64  
 6   Category             100000 non-null  str    
 7   LaunchMonth          100000 non-null  str    
 8   Country              100000 non-null  str    
 9   Currency             100000 non-null  str    
 10  OwnerExperience      100000 non-null  int64  
 11  VideoIncluded        100000 non-null  str    
 12  SocialMediaPresence  100000 non-null  int64  
 13  NumUpdates           100000 non-null  int64  
 14  IsSuccessful         100000 non-null  int64  
dtypes: float64(1), int64(8), str(

In [ ]:
print(df.describe())
#describe() → shows average, min, max values

          Unnamed: 0     GoalAmount   RaisedAmount   DurationDays  \
count  100000.000000  100000.000000  100000.000000  100000.000000   
mean    49999.500000   50538.001310   50598.121803      52.047370   
std     28867.657797   28613.822592   33255.277493      21.653238   
min         0.000000    1000.000000     528.632349      15.000000   
25%     24999.750000   25724.750000   23396.500017      33.000000   
50%     49999.500000   50706.000000   45980.674476      52.000000   
75%     74999.250000   75316.000000   72790.183158      71.000000   
max     99999.000000   99999.000000  149238.361559      89.000000   

          NumBackers  OwnerExperience  SocialMediaPresence     NumUpdates  \
count  100000.000000    100000.000000        100000.000000  100000.000000   
mean     1003.347340         9.489130         50077.301010      24.483230   
std       574.485153         5.763538         28767.521522      14.413833   
min        10.000000         0.000000           100.000000       0.000

In [11]:
#Step 3: Handle Missing Values
# Fill missing values
df = df.fillna(df.mean(numeric_only=True))

# For text columns
df = df.fillna("Unknown")

In [21]:
#Step 4: Remove Duplicate Rows
df = df.drop_duplicates()

| Action               | Output shown? | Why                              |
| -------------------- | ------------- | -------------------------------- |
| print(df.head())     | ✅ Yes         | showing data                     |
| df.fillna(...)       | ❌ No          | only modifying data              |
| df.drop_duplicates() | ❌ No          | cleaning data                    |
| df.mean()            | ❌ No          | calculation only used internally |


In [22]:
df_before = df.copy()

In [23]:
print("🔵 BEFORE ENCODING")
print(df_before.head())

🔵 BEFORE ENCODING
   Unnamed: 0  CampaignID  GoalAmount  RaisedAmount  DurationDays  NumBackers  \
0           0           0       76159  73518.028955            15        1903   
1           1           1       69394  60068.683288            49         469   
2           2           2       18652  22958.153470            58         148   
3           3           3       19487  24229.048390            39         780   
4           4           4       26734  17984.630311            68        1622   

   Category  LaunchMonth  Country  Currency  OwnerExperience  VideoIncluded  \
0         1           11        0         3               11              1   
1         0           10        0         3                7              1   
2         1            3        4         2               17              0   
3         1           10        2         0               10              0   
4         2           11        0         0               12              1   

   SocialMediaPresen

In [24]:
#STEP 3: Encode
from sklearn.preprocessing import LabelEncoder

le = LabelEncoder()

for col in df.columns:
    if df[col].dtype == 'object':
        df[col] = le.fit_transform(df[col].astype(str))

In [25]:
print("🟢 AFTER ENCODING")
print(df.head())

🟢 AFTER ENCODING
   Unnamed: 0  CampaignID  GoalAmount  RaisedAmount  DurationDays  NumBackers  \
0           0           0       76159  73518.028955            15        1903   
1           1           1       69394  60068.683288            49         469   
2           2           2       18652  22958.153470            58         148   
3           3           3       19487  24229.048390            39         780   
4           4           4       26734  17984.630311            68        1622   

   Category  LaunchMonth  Country  Currency  OwnerExperience  VideoIncluded  \
0         1           11        0         3               11              1   
1         0           10        0         3                7              1   
2         1            3        4         2               17              0   
3         1           10        2         0               10              0   
4         2           11        0         0               12              1   

   SocialMediaPresenc

In [26]:
print(df.select_dtypes(include='object').columns)

Index([], dtype='str')


In [27]:
print(df.head())

   Unnamed: 0  CampaignID  GoalAmount  RaisedAmount  DurationDays  NumBackers  \
0           0           0       76159  73518.028955            15        1903   
1           1           1       69394  60068.683288            49         469   
2           2           2       18652  22958.153470            58         148   
3           3           3       19487  24229.048390            39         780   
4           4           4       26734  17984.630311            68        1622   

   Category  LaunchMonth  Country  Currency  OwnerExperience  VideoIncluded  \
0         1           11        0         3               11              1   
1         0           10        0         3                7              1   
2         1            3        4         2               17              0   
3         1           10        2         0               10              0   
4         2           11        0         0               12              1   

   SocialMediaPresence  NumUpdates  Is

In [28]:
print(df.isnull().sum())

Unnamed: 0             0
CampaignID             0
GoalAmount             0
RaisedAmount           0
DurationDays           0
NumBackers             0
Category               0
LaunchMonth            0
Country                0
Currency               0
OwnerExperience        0
VideoIncluded          0
SocialMediaPresence    0
NumUpdates             0
IsSuccessful           0
dtype: int64


In [ ]:
#Perform feature selection
X = df.drop("IsSuccessful", axis=1)
y = df["IsSuccessful"]

In [30]:
print(X.head())

   Unnamed: 0  CampaignID  GoalAmount  RaisedAmount  DurationDays  NumBackers  \
0           0           0       76159  73518.028955            15        1903   
1           1           1       69394  60068.683288            49         469   
2           2           2       18652  22958.153470            58         148   
3           3           3       19487  24229.048390            39         780   
4           4           4       26734  17984.630311            68        1622   

   Category  LaunchMonth  Country  Currency  OwnerExperience  VideoIncluded  \
0         1           11        0         3               11              1   
1         0           10        0         3                7              1   
2         1            3        4         2               17              0   
3         1           10        2         0               10              0   
4         2           11        0         0               12              1   

   SocialMediaPresence  NumUpdates  
0

In [32]:
#Split the dataset into training and testing sets
from sklearn.model_selection import train_test_split

In [33]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

In [34]:
print(X_train.shape, X_test.shape)

(80000, 14) (20000, 14)


| Part        | Meaning                                 |
| ----------- | --------------------------------------- |
| (80000, 14) | Training data → 80,000 rows, 14 columns |
| (20000, 14) | Testing data → 20,000 rows, 14 columns  |


In [35]:
from sklearn.preprocessing import StandardScaler

In [36]:
scaler = StandardScaler()

In [37]:
X_train = scaler.fit_transform(X_train)

In [38]:
X_test = scaler.transform(X_test)

In [41]:
from sklearn.linear_model import LogisticRegression

# Create model
model = LogisticRegression()

# Train model
model.fit(X_train, y_train)

# Make predictions
y_pred = model.predict(X_test)

# Show sample predictions
print(y_pred[:10])

[0 0 1 0 0 1 0 1 0 1]


| Value | Meaning                 |
| ----- | ----------------------- |
| 0     | Campaign Unsuccessful ❌ |
| 1     | Campaign Successful ✅   |


In [42]:
from sklearn.metrics import accuracy_score

print(accuracy_score(y_test, y_pred))

0.9989


#Task 2 – Classification Model Building

In [44]:
#Random Forest ONLY
from sklearn.ensemble import RandomForestClassifier

rf = RandomForestClassifier(n_estimators=100, random_state=42)
rf.fit(X_train, y_train)
rf_pred = rf.predict(X_test)

print(rf_pred[:10])

[0 0 0 0 0 1 0 1 0 1]


In [45]:
#KNN ONLY
from sklearn.neighbors import KNeighborsClassifier

knn = KNeighborsClassifier(n_neighbors=5)
knn.fit(X_train, y_train)
knn_pred = knn.predict(X_test)

print(knn_pred[:10])

[0 0 0 0 0 1 0 0 1 0]


In [46]:
#Decision Tree
from sklearn.tree import DecisionTreeClassifier

dt = DecisionTreeClassifier(max_depth=5, random_state=42)
dt.fit(X_train, y_train)
dt_pred = dt.predict(X_test)

print(dt_pred[:10])

[0 0 0 0 0 1 0 1 0 1]


In [47]:
#NAIVE BAYES ONLY
from sklearn.naive_bayes import GaussianNB

nb = GaussianNB()
nb.fit(X_train, y_train)
nb_pred = nb.predict(X_test)

print(nb_pred[:10])

[0 0 0 0 0 1 0 1 0 0]


In [49]:
#Support Vector Machine (SVM)
from sklearn.svm import SVC

svm = SVC(kernel='linear', C=1)
svm.fit(X_train, y_train)
svm_pred = svm.predict(X_test)

print(svm_pred[:10])

[0 0 1 0 0 1 0 1 0 1]


In [ ]:
from sklearn.metrics import accuracy_score
#Now we compare models using accuracy scores.
print("Random Forest:", accuracy_score(y_test, rf_pred))
print("KNN:", accuracy_score(y_test, knn_pred))
print("Decision Tree:", accuracy_score(y_test, dt_pred))
print("Naive Bayes:", accuracy_score(y_test, nb_pred))
print("SVM:", accuracy_score(y_test, svm_pred))

Random Forest: 0.9645
KNN: 0.69445
Decision Tree: 0.83735
Naive Bayes: 0.6619
SVM: 0.99905


| Model         | Accuracy           |
| ------------- | ------------------ |
| Random Forest | 0.9645 (96.45%)    |
| KNN           | 0.69445 (69.44%)   |
| Decision Tree | 0.83735 (83.73%)   |
| Naive Bayes   | 0.6619 (66.19%)    |
| SVM           | 0.99905 (99.90%) ⭐ |


Best Model = SVM (Support Vector Machine)

It gives the highest accuracy: 99.91%.
SVM performed the best among all the classification models with an accuracy of 99.91%, indicating strong classification performance for crowdfunding success prediction.

In [6]:
#Model Evaluation
from sklearn.metrics import accuracy_score
from sklearn.metrics import precision_score
from sklearn.metrics import recall_score
from sklearn.metrics import f1_score

